src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -14,6 +14,7 @@ public static class ValidationUtilities
     {
         private static readonly string XmsPageable = "x-ms-pageable";
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private static readonly Regex resourceProviderPathPattern = new Regex(@"/providers/(?<resPath>[^{/]+)/", RegexOptions.IgnoreCase);
```

Comment:
You are right. The current implementation is just checking whether we have any path that ends with `/operations` and that can bring false positive as we'd want to flag out the once that ending with `providers/{name}/operations` and technically we can have more than one providers in the same swagger (although we do not want that and we have other rule that catches that but at the same time we do not want to miss out when we have more than one provider and only one of them have /providers/{name}/operations). 

We can make this simple to just catch for one provider and assume that as user fixes the swagger we'll surface other errors. It's upto us I am find with both approach :)